proposals/exception-handling/Exceptions-formal-overview.md

```diff
@@ -0,0 +1,201 @@
+# 3rd proposal formal spec overview
+
+This is an overview of the 3rd proposal's formal spec additions, to aid in discussions concerning the proposed semantics.
+
+## Abstract Syntax
+
+### Types
+
+Tag Types
+
+```
+tagtype ::= [t*]→[]
+```
+
+### Instructions
+
+```
+instr ::= ... | throw x | rethrow l
+        | try bt instr* (catch x instr*)* (catch_all instr*)? end
+        | try bt instr* delegate l
+```
+
+### Modules
+
+Tags
+
+```
+tag ::= export* tag tagtype  | export* tag tagtype import
+```
+
+Modules
+
+```
+mod ::= module ... tag*
+```
+
+## Validation (Typing)
+
+#### Modification to labels
+
+To verify that a `try...delegate l` instruction refers to a label surrounding the instructions of a try block (call this a try-label), introduce a `kind` attribute to labels in the validation context, which is set to `try` when the label is a try-label.
+
+Similarly, to verify that the `rethrow l` instruction refers to a label surrounding the instructions of a catch block (call this a catch-label), we allow the `kind` attribute of labels in the validation context to be set to `catch` when the label is a catch-label.
```

Comment:
To reply for @ioannad: during validation, a try label is used within the try body, a catch label within a catch clause. The former can be delegated to, the latter rethrown from. (Furthermore, if we picked option (3) for delegate, we wouldn't need to distinguish try labels.)

Reply:
Exactly what @rossberg said. I am also fond of option 3 btw, thinking it will simplify the spec. As a followup to this PR I could write the formal overview for option 3. It should be fairly straightforward, but unfortunately I won't have any free time for this during this week.